.github/tests/common.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+get_namespace_details () {
+cat <<EOF >>"$GITHUB_STEP_SUMMARY"
+### Namespace $1
+
+#### Events
+
+\`\`\`shell
+$(kubectl --request-timeout=30s get events --output wide --namespace "$1")
+\`\`\`
+
+#### Pods
+
+\`\`\`shell
+$(kubectl --request-timeout=30s describe pods --namespace "$1")
+\`\`\`
+
+#### Logs
+
+\`\`\`shell
+$(kubectl get pods -o name -n "$1" | while read -r line; do echo logs for "${line}"; kubectl logs -n "$1" "${line}" --all-containers=true --ignore-errors=true; done)
+\`\`\`
+
+EOF
+}
+
+k_wait () {
+  kubectl wait --for condition=available --timeout 30s --namespace "$1" "$2" "$3" | tail -n 1
+}
+
+k_rollout_status () {
+  kubectl rollout status --watch --timeout 30s --namespace "$1" "$2" "$3" | tail -n 1
+}
+
+print_spire_workload_status () {
+  cat <<EOF >>"$GITHUB_STEP_SUMMARY"
+### $1
+
+| workload                             | Status |
+| ------------------------------------ | ------ |
+| spire-server                         | <pre>$(k_rollout_status "$1" statefulset spire-server)</pre> |
+| spire-controller-manager             | <pre>$(k_rollout_status "$1" statefulset spire-controller-manager)</pre> |
+| spire-spiffe-csi-driver              | <pre>$(k_rollout_status "$1" daemonset spire-spiffe-csi-driver)</pre> |
+| spire-agent                          | <pre>$(k_rollout_status "$1" daemonset spire-agent)</pre> |
+| spire-spiffe-oidc-discovery-provider | <pre>$(k_rollout_status "$1" deployments.apps spire-spiffe-oidc-discovery-provider)</pre> |
```

Comment:
These all use $1, assuming there is only ever one namespace. Might be more reusable to define $1 and $2 so split namespaces work too?